### PR TITLE
adds use of the IOMMU with the ixy.rs driver

### PR DESCRIPTION
**Pull request by huberste**

### CODE REVIEW

**Review by ackxolotl — CHANGES_REQUESTED** [review]
Since `IxgbeIommuDevice` is just a wrapper for `IxgbeDevice` with no big differences to `IxgbeDevice` apart from the initialisation procedure I think it would be better to add that code to `IxgbeDevice` in a method like `init_iommu()` and to move the check if iommu is available from `ixy_init()` in `lib.rs` to the `init()` method of `IxgbeDevice`. If all that code is in in `ixgbe.rs` then it is no longer necessary to make `IxgbeRxQueue`, `IxgbeTxQueue` and some fields in `IxgbeDevice` public. Also, what happened to `echoer.rs`? :D

File: src/ixgbe.rs [hunk 1]
```diff
@@ -102,6 +108,21 @@ impl IxyDevice for IxgbeDevice {
         DRIVER_NAME
     }
 
+    /// Returns the driver's iommu capability.
+    fn is_card_iommu_capable(&self) -> bool {
+        self.iommu
+    }
+
+    /// Returns the VFIO container file descriptor.
+    /// When implementing non-VFIO / IOMMU devices, just return 0.
```
Comment (ackxolotl) [inline src/ixgbe.rs]: Returns VFIO container file descriptor or [\`None\`] if IOMMU is not available.

**Comment by huberste** [comment]
I implemented the changes you requested and also unified the comment style (and fixed some commentation).

**Comment by huberste** [comment]
I implemented the changes you requested and also unified the comment style (and fixed some commentation).

**Review by ackxolotl — APPROVED** [review]

**Pull request merged**